lib/irb/completion.rb

```diff
@@ -137,27 +137,34 @@ def doc_namespace(preposing, matched, _postposing, bind:)
   end
 
   class RegexpCompletor < BaseCompletor # :nodoc:
+    KERNEL_METHODS = ::Kernel.instance_method(:methods)
+    KERNEL_PRIVATE_METHODS = ::Kernel.instance_method(:private_methods)
+    KERNEL_INSTANCE_VARIABLES = ::Kernel.instance_method(:instance_variables)
+    KERNEL_GLOBAL_VARIABLES = ::Kernel.instance_method(:global_variables)
+    OBJECT_CLASS_INSTANCE_METHOD = ::Object.instance_method(:class)
+    MODULE_CONSTANTS_INSTANCE_METHOD = ::Module.instance_method(:constants)
+
     using Module.new {
       refine ::Binding do
         def eval_methods
-          ::Kernel.instance_method(:methods).bind(eval("self")).call
+          KERNEL_METHODS.bind(eval("self")).call
         end
 
         def eval_private_methods
-          ::Kernel.instance_method(:private_methods).bind(eval("self")).call
+          KERNEL_PRIVATE_METHODS.bind(eval("self")).call
         end
 
         def eval_instance_variables
-          ::Kernel.instance_method(:instance_variables).bind(eval("self")).call
+          KERNEL_INSTANCE_VARIABLES.bind(eval("self")).call
         end
 
         def eval_global_variables
-          ::Kernel.instance_method(:global_variables).bind(eval("self")).call
+          KERNEL_GLOBAL_VARIABLES.bind(eval("self")).call
```

Comment:
`::Kernel.global_variables` is better because this does not need to look into `self`